dpd/src/api_server.rs

```diff
@@ -3414,148 +2020,77 @@ mod imp {
-    api.register(link_fault_get).unwrap();
-    api.register(link_fault_clear).unwrap();
-    api.register(link_fault_inject).unwrap();
-    api.register(link_ipv4_list).unwrap();
-    api.register(link_ipv4_create).unwrap();
-    api.register(link_ipv4_reset).unwrap();
-    api.register(link_ipv4_delete).unwrap();
-    api.register(link_ipv6_list).unwrap();
-    api.register(link_ipv6_create).unwrap();
-    api.register(link_ipv6_delete).unwrap();
-    api.register(link_ipv6_reset).unwrap();
-    api.register(link_nat_only_set).unwrap();
-    api.register(link_nat_only_get).unwrap();
-    api.register(link_mac_get).unwrap();
-    // TODO-correctness: A link's MAC address should be determined by the FRUID
-    // data, not under the control of the client. We really only need this for
-    // the integration tests in `dpd-client`. We should consider removing it for
-    // production.
-    api.register(link_mac_set).unwrap();
-    api.register(link_history_get).unwrap();
-
-    api.register(loopback_ipv4_list).unwrap();
-    api.register(loopback_ipv4_create).unwrap();
-    api.register(loopback_ipv4_delete).unwrap();
-    api.register(loopback_ipv6_list).unwrap();
-    api.register(loopback_ipv6_create).unwrap();
-    api.register(loopback_ipv6_delete).unwrap();
-
-    api.register(nat_ipv6_addresses_list).unwrap();
-    api.register(nat_ipv6_list).unwrap();
-    api.register(nat_ipv6_get).unwrap();
-    api.register(nat_ipv6_create).unwrap();
-    api.register(nat_ipv6_delete).unwrap();
-    api.register(nat_ipv6_reset).unwrap();
-    api.register(nat_ipv4_addresses_list).unwrap();
-    api.register(nat_ipv4_list).unwrap();
-    api.register(nat_ipv4_get).unwrap();
-    api.register(nat_ipv4_create).unwrap();
-    api.register(nat_ipv4_delete).unwrap();
-    api.register(nat_ipv4_reset).unwrap();
-    api.register(oximeter_collect_meta_endpoint).unwrap();
-
-    api.register(port_settings_apply).unwrap();
-    api.register(port_settings_clear).unwrap();
-    api.register(port_settings_get).unwrap();
-    api.register(switch_identifiers).unwrap();
-    api.register(tfport_data).unwrap();
-
-    api.register(ipv4_nat_generation).unwrap();
-    api.register(ipv4_nat_trigger_update).unwrap();
-
-    api.register(multicast_group_create).unwrap();
-    api.register(multicast_group_create_external).unwrap();
-    api.register(multicast_reset).unwrap();
-    api.register(multicast_group_delete).unwrap();
-    api.register(multicast_group_update).unwrap();
-    api.register(multicast_group_update_external).unwrap();
-    api.register(multicast_group_get).unwrap();
-    api.register(multicast_groups_list).unwrap();
-    api.register(multicast_groups_list_by_tag).unwrap();
-    api.register(multicast_reset_by_tag).unwrap();
-    api.register(multicast_reset_untagged).unwrap();
+    #[allow(unused_mut)]
+    let mut api = dpd_api_mod::api_description::<DpdApiImpl>().unwrap();
 
+    // TODO: need to move these into dpd-api
```

Comment:
Am I reading this correctly that the API now has trait-based parts and non-trait based parts. E.g all the endpoints that were previously registered above are now in the trait, and the ASIC-specific endpoints below are still in the API but not in the trait?

Reply:
Correct, yeah. At the moment the function-based endpoints are part of the OpenAPI document, though with the OpenAPI manager we'll need all endpoints present in the document to be moved into the trait. Hence the TODO.